### PR TITLE
ADEN-12473 - Added integration with Captify

### PR DESCRIPTION
**Pull request by domjanik**

### CODE REVIEW

File: @types/window/index.d.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ interface Window {
 	au_seg?: any;
 	beaconId?: string;
 	beacon_id?: string;
+	captify_kw_query_12974: string;
```
Comment (mfarul) [inline @types/window/index.d.ts:19]: will be always available? eg ?:

File: src/ad-services/captify/index.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { BaseServiceSetup, context, utils } from '@ad-engine/core';
+
+const logGroup = 'captify';
+
+class Captify extends BaseServiceSetup {
+	PIXEL_URL = 'https://p.cpx.to/p/12974/px.js';
+
+	isEnabled(): boolean {
```
Comment (mfarul) [inline src/ad-services/captify/index.ts:8]: i see this so many times .... maybe a base class method with service variable param ?

File: src/ad-services/captify/index.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	async call(): Promise<void> {
+		if (!this.isEnabled()) {
+			utils.logger(logGroup, 'disabled');
+
+			return Promise.resolve();
+		}
+		window.captify_kw_query_12974 = '';
+		const elem = document.createElement('script');
```
Comment (mfarul) [inline src/ad-services/captify/index.ts:24]: can we use ScriptLoader.createScript ?

**Review by mfarul — APPROVED** [review]

File: src/ad-services/captify/index.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		);
+	}
+
+	async call(): Promise<void> {
+		if (!this.isEnabled()) {
+			utils.logger(logGroup, 'disabled');
+
+			return Promise.resolve();
+		}
+		window.captify_kw_query_12974 = '';
```
Comment (mfarul) [inline src/ad-services/captify/index.ts:23]: do we need to set this? Captify script does not handle this on their own ?

Comment (domjanik) [reply src/ad-services/captify/index.ts:23]: According to Captify's documentation we got we should set this by ourself :(

**Pull request merged**